core/block.js

```diff
@@ -650,7 +650,17 @@ Blockly.Block.prototype.setDeletable = function(deletable) {
  */
 Blockly.Block.prototype.isMovable = function() {
   return this.movable_ && !this.isShadow_ &&
-      !(this.workspace && this.workspace.options.readOnly);
+    !(this.workspace && 
```

Comment:
Should probably just call `isMovablePersisted()` instead of duplicating the logic. Same with `isEditable()` below